### PR TITLE
Add a Docker-based setup.

### PR DESCRIPTION
**Pull request by sjrd**
I can't run Ruby 2.7.x on my machine anymore. Docker was the only solution I found. It may help others getting started to contribute to this website as well.

The setup is basically copied from that of `scala-lang.org`, but with an additional setup of Node.js.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version: '2'
+
+services:
+  jekyll:
+    user: "${UID}:${GID}"
+    build: .
+    command: sh -c "chown $UID / && echo NODE && node -v && bundle exec jekyll serve --incremental --host=0.0.0.0 "
```
Comment (gzm0) [inline docker-compose.yml:7]: Do we need `--watch` here (like the README.md says)?

**Review by gzm0 — APPROVED** [review]
A bunch of suggestions for improvements, but for this kind of build, not super critical.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+RUN apt install -y nodejs
+
+RUN gem install bundler:2.3.10
+
+WORKDIR /srv/jekyll
+
+COPY Gemfile .
+COPY Gemfile.lock .
+
+RUN echo -n "bundle version: " && bundle --version
```
[the remark this answers is not in the document]
Comment (sjrd) [reply Dockerfile:14]: Yes, indeed. I guess this was a debub print.

File: docker-compose.yml (hunk 1 above)
Comment (sjrd) [reply docker-compose.yml:7]: That would be more consistent, indeed.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM ruby:2.7.5
+
+RUN apt install -y curl
+RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash -
+RUN apt install -y nodejs
```
[the remark this answers is not in the document]
Comment (sjrd) [reply Dockerfile:6]: Yes. We need it for one the plugins, which calls into JS for some reason. I added a comment.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    ports:
+      - '4000:4000'
+    volumes:
+      - .:/srv/jekyll
```
[the remark this answers is not in the document]
Comment (gzm0) [reply docker-compose.yml:11]: :) that's pretty much how this looked 😹

If this is not super urgent, I can make an alternative suggestion (I use docker daily now).

If yes, I'd appreciate a quick description of your deployment process to make sure I'm not making it worse.

[the remark this answers is not in the document]
Comment (sjrd) [reply docker-compose.yml:11]: It's not urgent at all. I only use it locally. I can keep it local until there's something good.

The deployment is still working Docker-free in GitHub Actions. It's in https://github.com/scala-js/scala-js-website/blob/main/.github/workflows/deploy.yml

**Comment by gzm0** [comment]
Alternative: #601.

**Comment by sjrd** [comment]
Superseded by #601.

**Pull request closed without merging**